sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/ExpressionCompiler.java

```diff
@@ -682,12 +682,29 @@ node, new DBSPTypeBool(CalciteObject.EMPTY, false), DBSPOpcode.EQ,
                     }
                     case "numeric_inc":
                     case "sign":
-                    case "log10":
-                    case "ln":
                     case "abs": {
                         return this.compilePolymorphicFunction(call, node, type,
                                 ops, 1);
                     }
+                    case "log10":
+                    case "ln":
+                    {
+                        // Cast to Double if the type isn't Decimal
+                        DBSPExpression arg = ops.get(0);
+                        if (!arg.type.is(DBSPTypeDecimal.class)) {
+                            ops.set(0, arg.cast(new DBSPTypeDouble(node, arg.getType().mayBeNull)));
+                        }
+                        return this.compilePolymorphicFunction(call, node, type, ops, 1);
+                    }
+                    case "log":
+                    {
+                        // Turn the arguments into Double
+                        for (int i = 0; i < ops.size(); i++) {
+                            DBSPExpression arg = ops.get(i);
+                            ops.set(i, arg.cast(new DBSPTypeDouble(node, arg.getType().mayBeNull)));
+                        }
+                        return this.compilePolymorphicFunction(call, node, type, ops, 1, 2);
+                    }
```

Comment:
We only support log for double as rust_decimal doesn't have a native decimal support for it.